DevEdu.API/DevEdu.API/Controllers/UserController.cs

```diff
@@ -1,47 +1,59 @@
-﻿using DevEdu.API.Models.InputModels;
+﻿using AutoMapper;
+using DevEdu.API.Models.InputModels;
+using DevEdu.DAL.Models;
+using DevEdu.DAL.Repositories;
 using Microsoft.AspNetCore.Mvc;
+using System.Collections.Generic;
 
 namespace DevEdu.API.Controllers
 {
     [ApiController]
     [Route("api/[controller]")]
     public class UserController : Controller
     {
-        public UserController() { }
+        UserRepository _userRepository;
+        private readonly IMapper _mapper;
+
+        public UserController(Mapper mapper) {
+            _mapper = mapper;
+            _userRepository = new UserRepository();
+        }
 
         // api/user
         [HttpPost]
         public int AddUser([FromBody] UserInsertInputModel model)
         {
-            return 1;
+            var dto = _mapper.Map<UserDto>(model);
+            return _userRepository.AddUser(dto);
         }
 
         // api/user/userId
         [HttpPut("{userId}")]
-        public string UpdateUserById(int userId,[FromBody] UserUpdateInputModel model)
+        public void UpdateUserById(int userId, [FromBody] UserUpdateInputModel model)
         {
-            return $"update User №{userId} is success";
+            var dto = _mapper.Map<UserDto>(model);
+            _userRepository.UpdateUser(userId, dto);

```

Comment:
this method is supposed to take only one argument (dto). please change it accordingly